modules/gdscript/gdscript_analyzer.cpp

```diff
@@ -1905,17 +1905,14 @@ void GDScriptAnalyzer::resolve_return(GDScriptParser::ReturnNode *p_return) {
 	GDScriptParser::DataType result;
 
 	GDScriptParser::DataType expected_type;
-	bool has_expected_type = false;
-
-	if (parser->current_function != nullptr) {
+	bool has_expected_type = parser->current_function != nullptr;
+	if (has_expected_type) {
 		expected_type = parser->current_function->get_datatype();
-		has_expected_type = true;
 	}
 
 	if (p_return->return_value != nullptr) {
 		reduce_expression(p_return->return_value);
 		if (p_return->return_value->type == GDScriptParser::Node::ARRAY) {
-			// Check if assigned value is an array literal, so we can make it a typed array too if appropriate.
```

Comment:
Any reason to remove this comment?

Reply:
Comment does not add anything of value in my opinion, in array PR it is pretty obvious what the thing does:
```cpp
if (p_return->return_value->type == GDScriptParser::Node::ARRAY && has_expected_type && expected_type.has_container_element_type()) {
  update_array_literal_element_type(static_cast<GDScriptParser::ArrayNode *>(p_return->return_value), expected_type.get_container_element_type());
}
```
If it is not clear than maybe the name of the function should be changed.

Reply:
I'd keep the comment

Reply:
I put the comment back. 

`update_array_literal_element_type` has only one function - to update an array literal type to a specified type, that's it. Commented code is a single `if` block with a single line with that method, there's no mystery as to when, why and what it does.

But ok.